mitodl/open-discussions
# View/edit profile

#### What are the relevant tickets?
- Closes #683, #783, #784

#### What's this PR do?
- Adds pages & form for viewing and editing a profile

#### How should this be manually tested?
- When not logged in, you should see a default profile image at top right, clicking on it will not provide an option to 'View Profile'
- When logged in, you should see at top right either:
  - your profile image if you have one from Micromasters
  - a circle with your initials on it if you don't have a profile image
  - clicking on the above should provide a 'View Profile' option.
- Click 'View Profile'
- You should see a card with your profile info (name, bio, headline) including your profile image (or initials).  There should be a little circle with either an 'add' (+) or or 'edit' (pencil) icon at upper right of the image/circle, depending on whether you have an image or not.
- Clicking the add/edit icon should bring up the image selector/cropper, and using it should update your profile image.
- Click the 'Edit Profile' button.  The profile image should appear as before, and now your other info should be in input fields.
- If you clear out your name and submit, you should get an error message indicating it's required.
- Add/change your name, bio, headline and click submit.  You should be brought back to the 'View Profile' page with the updated info displayed.
- Find the username of another user and go to `/profiles/<username>/edit`.  You should be redirected back to the 'View Profile' page, which should display that user's profile info.  There should not be an add/edit icon for the profile image.

#### Where should the reviewer start?
https://invis.io/ZJKZT2725DF#/301006569_Edit_Profile
